packages/serialization/form/src/formSerializationWriter.ts

```diff
@@ -0,0 +1,168 @@
+/* eslint-disable @typescript-eslint/no-unused-expressions */
+import {
+  DateOnly,
+  Duration,
+  Parsable,
+  SerializationWriter,
+  TimeOnly,
+} from "@microsoft/kiota-abstractions";
+
+export class FormSerializationWriter implements SerializationWriter {
+  private readonly writer: string[] = [];
+  private static propertySeparator = `&`;
+  private depth = -1;
+  public onBeforeObjectSerialization: ((value: Parsable) => void) | undefined;
+  public onAfterObjectSerialization: ((value: Parsable) => void) | undefined;
+  public onStartObjectSerialization:
+    | ((value: Parsable, writer: SerializationWriter) => void)
+    | undefined;
+  public writeStringValue = (key?: string, value?: string): void => {
```

Comment:
If this is called with a falsy key and a truthy value then the value is written without a key or separator, is that valid?

Reply:
thanks for calling this out, solved.